Tella/Scenes/Settings/Views/Servers/ServerSelectionView.swift

```diff
@@ -77,6 +80,12 @@ struct ServerSelectionView: View {
             .environmentObject(serverViewModel)
             .environmentObject(serversViewModel))
     }
+    
+    fileprivate func navigateToGDriveFlow() {
+        gDriveVM.handleSignIn { 
+            navigateTo(destination: SelectDriveConnection( dGriveServerViewModel: GDriveServerViewModel()), title: "Select Google drive")
+        }
+    }
 
```

Comment:
You can replace it by this instead of using completions :
   ```
 fileprivate func navigateToGDriveFlow()   {
        Task {
            do {
                try await gDriveVM.handleSignIn()
                DispatchQueue.main.async {
                    navigateTo(destination: SelectDriveConnection( dGriveServerViewModel: GDriveServerViewModel()), title: "Select Google drive")
                }
            }
            catch{
                Toast.displayToast(message: error.localizedDescription)
            }
        }
    }

```

Reply:
if i'm not mistaken the `navigateTo` methodis part of the view layer and manages the UI navigation. I think we are doing this to maintain a separation of responsibilities, so we keep all the navigation logic within the views.

but I get your point! so we could adjust the method to avoid using completion handlers and also handle the toast from the view:
```
fileprivate func navigateToGDriveFlow() {
    Task {
        do {
            try await gDriveVM.handleSignIn()
            DispatchQueue.main.async {
                navigateTo(
                    destination: SelectDriveConnection(gDriveServerViewModel: GDriveServerViewModel(mainAppModel: mainAppModel)),
                    title: LocalizableSettings.settServerGDrive.localized
                )
            }
        } catch {
            DispatchQueue.main.async {
                Toast.displayToast(message: error.localizedDescription)
            }
        }
    }
}
```

Reply:
Yes, we should have the state and try catch should be sent from the VM and handle the navigations and Toast appearance  in the View